### PR TITLE
Stats core changes

### PR DESCRIPTION
**Pull request by MarceloAquino7**
This PR contains all changes inside stats core in order to support exporters and the new transport class.

### CODE REVIEW

**Comment by MarceloAquino7** [comment]
@songy23 I think that everything is fine right now.

File: opencensus/common/transports/async.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+	@property
+	def is_alive(self):
+		"""Returns True is the background thread is running."""
+		return self._thread is not None and self._thread.is_alive()
+
+	def _get_items(self):
+		"""Get multiple items from a Queue.
+
+		Gets at least one (blocking) and at most ``max_items`` items
```
Comment (odeke-em) [inline opencensus/common/transports/async.py]: `max_items` isn't defined anywhere. Did we perhaps mean to use `max_batch_size`?

File: opencensus/common/transports/async.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+		return items
+
+	def _thread_main(self):
+		"""The entry point for the worker thread.
+
+		Pulls pending data off the queue and writes them in
+		batches to the specified tracing backend using the exporter.
+		"""
+		print('Background thread started.')
```
Comment (odeke-em) [inline opencensus/common/transports/async.py]: Perhaps this was a legacy print but it pollutes user programs, please remove it or perhaps make it a log that can be turned on

File: opencensus/common/transports/async.py (hunk 1 above)
Comment (MarceloAquino7) [reply opencensus/common/transports/async.py]: Yes, it was a mistake inside the comment. I'll fix that, thanks !

File: opencensus/common/transports/async.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+		"""Signals the background thread to stop.
+
+		This does not terminate the background thread. It simply queues the
+		stop signal. If the main process exits before the background thread
+		processes the stop signal, it will be terminated without finishing
+		work. The ``grace_period`` parameter will give the background
+		thread some time to finish processing before this function returns.
+
+		:type grace_period: float
+		:param grace_period: If specified, this method will block up to this
```
[the remark this answers is not in the document]
Comment (MarceloAquino7) [reply opencensus/common/transports/async.py]: Right now we are reading its value inside the constructor, so of course do not make sense this piece of comment. I'll remove it, thanks !

File: opencensus/common/transports/async.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+_DEFAULT_GRACE_PERIOD = 5.0  # Seconds
+_DEFAULT_MAX_BATCH_SIZE = 10
+_WAIT_PERIOD = 1.0  # Seconds
+_WORKER_THREAD_NAME = 'opencensus.common.Worker'
+_WORKER_TERMINATOR = object()
+
+
+class _Worker(object):
+	"""A background thread that exports batches of data.
```
Comment (liyanhui1228) [inline opencensus/common/transports/async.py]: The indentation of this file is different than other files in this repo, shall we change to use 4 spaces indentation to match with the repo?

File: opencensus/common/transports/base.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    def export(self, datas):
+        """Export the data."""
+        raise NotImplementedError
+
+    def flush(self):
+        """Submit any pending data.
+
+        For blocking/sync transports, this is a no-op.
+        """
+
```
Comment (liyanhui1228) [inline opencensus/common/transports/base.py]: Remove extra blank line.

Comment (MarceloAquino7) [reply opencensus/common/transports/base.py]: Thanks for your reply @liyanhui1228 , I will work on these items. About the test coverage, do you guys use 4 spaces for indentation ?

Comment (liyanhui1228) [reply opencensus/common/transports/base.py]: Yes we use 4 spaces indentation.

**Comment by odeke-em** [comment]
@liyanhui1228 they've updated the code with our requests, PTAL again :)

@MarceloAquino7 please rebase from master, as Github currently reports some conflicts with master.

**Comment by liyanhui1228** [comment]
This LGTM, will approve once the CI is green.

**Comment by odeke-em** [comment]
@liyanhui1228 oh cool, I've updated the branch and now CI is running, please check back in say 10 minutes for when CI returns.

**Comment by odeke-em** [comment]
@liyanhui1228 for some reason CircleCI tripped claiming that the coverage for some test failed
```shell
nox > Command coverage report --show-missing --fail-under=100 failed with exit code 2
```
https://circleci.com/gh/census-instrumentation/opencensus-python/616?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link PTAL

**Comment by liyanhui1228** [comment]
Yes we should add more test to fix that test coverage.

File: tests/unit/common/transports/test_sync.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    def test_export(self):
+        exporter = mock.Mock()
+        transport = sync.SyncTransport(exporter)
+        data = {
+            'traceId': 'test1',
+            'spans': [{}, {}],
+        }
+        transport.export(data)
+        self.assertTrue(True)
```
Comment (c24t) [inline tests/unit/common/transports/test_sync.py]: What happened here?

**Pull request merged**